core/src/main/java/com/yahoo/oak/InternalOakBasics.java

```diff
@@ -471,7 +456,8 @@ protected void invalidatePrevState() {
 
         protected abstract BasicChunk<K, V> getNextChunk(BasicChunk<K, V> current);
 
-        protected abstract BasicChunk.BasicChunkIter getChunkIter(BasicChunk<K, V> current);
+        protected abstract BasicChunk.BasicChunkIter getChunkIter(BasicChunk<K, V> current)
```

Comment:
It would be good to explain in which cases the exception is thrown. As I remember we were asking ourselves as well. Please add a comment.

Reply:
to be discussed